pytorch_lightning/utilities/cli.py

```diff
@@ -37,7 +37,8 @@
 
     set_config_read_mode(fsspec_enabled=True)
 else:
-    ArgumentParser = Namespace = object
+    locals()['ArgumentParser'] = object
```

Comment:
Wouldn't it be better to encapsulate this in a try catch? It will be more readable. This is working, but it's a bit complex?

Something like this:

```py
try:
    from jsonargparse import ActionConfigFile, ArgumentParser, class_from_function, Namespace, set_config_read_mode
    from jsonargparse.optionals import import_docstring_parse

    set_config_read_mode(fsspec_enabled=True)
except Exception as e: # this can be a better exception, just on top of my head
    if _JSONARGPARSE_AVAILABLE:
       raise e
    else:
       ArgumentParser = object
```

Reply:
Having `ArgumentParser = object` means that intellisense will not work, which is the purpose of this pull request. Does it work for you like this?

The purpose of the if is not catching an exception. Having `_JSONARGPARSE_AVAILABLE=True` already means that jsonargparse is installed. If installed and there is an exception while importing, it would be some issue in the environment (not an issue of cli.py module) and it is fine that it fails. No need to catch it.

Reply:
You are correct. Thanks!